influxql/ast.go

```diff
@@ -908,11 +908,34 @@ func (s *ShowSeriesStatement) RequiredPrivileges() ExecutionPrivileges {
 
 // DropSeriesStatement represents a command for removing a series from the database.
 type DropSeriesStatement struct {
-	Name string
+	// The Id of the series being dropped (optional)
+	SeriesID uint32
+
+	// Data source that fields are extracted from (optional)
+	Source Source
+
+	// An expression evaluated on data point (optional)
+	Condition Expr
 }
 
 // String returns a string representation of the drop series statement.
-func (s *DropSeriesStatement) String() string { return fmt.Sprintf("DROP SERIES %s", s.Name) }
+func (s *DropSeriesStatement) String() string {
+	var buf bytes.Buffer
+	_, _ = buf.WriteString("DROP SERIES ")
+
+	if s.Source != nil {
+		_, _ = buf.WriteString("FROM ")
+		_, _ = buf.WriteString(s.Source.String())
+		if s.Condition != nil {
```

Comment:
The `if s.Condition` block should be at the same level as the `if s.Source` because you can have either of them with or without the other.  e.g.,...
```sql
DROP SERIES FROM cpu;

DROP SERIES WHERE region = 'uswest';

DROP SERIES FROM cpu WHERE region = 'uswest';
```